src/ManagerV2/DocumentManager.php

```diff
@@ -42,34 +41,42 @@ public function __construct(
     /**
      * @return Document[]
      */
-    public function getAllDocumentsWithUrl(Beneficiaire $beneficiary, ?Dossier $folder = null): array
+    public function getDocumentsWithUrl(Beneficiaire $beneficiary, ?Dossier $folder = null): array
     {
-        return $this->getDocumentsWithUrl($this->repository->findAllByBeneficiary($beneficiary, $folder));
+        return $this->hydrateDocumentsAndThumbnailWithUrl(
+            $this->getUser() === $beneficiary->getUser()
+                ? $this->repository->findAllByBeneficiary($beneficiary, $folder)
+                : $this->repository->findSharedByBeneficiary($beneficiary, $folder),
+        );
     }
 
     /**
      * @return Document[]
      */
-    public function getSharedDocumentsWithUrl(Beneficiaire $beneficiary, ?Dossier $folder = null): array
+    public function searchDocumentsWithUrl(Beneficiaire $beneficiary, ?string $search, Dossier $folder = null): array
     {
-        return $this->getDocumentsWithUrl($this->repository->findSharedByBeneficiary($beneficiary, $folder));
+        return $this->hydrateDocumentsAndThumbnailWithUrl(
+            $beneficiary->getUser() === $this->getUser()
+                ? $this->repository->searchByBeneficiary($beneficiary, $search, $folder)
+                : $this->repository->searchSharedByBeneficiary($beneficiary, $search, $folder),
+        );
     }
 
     /**
      * @param Document[] $documents
      *
      * @return Document[]
      */
-    public function getDocumentsWithUrl(array $documents): array
+    public function hydrateDocumentsAndThumbnailWithUrl(array $documents): array
     {
         foreach ($documents as $document) {
-            $this->getDocumentWithUrl($document);
+            $this->hydrateDocumentAndThumbnailWithUrl($document);
```

Comment:
maybee you can shortcut that with a map
return array_map(fn(Document $document) => $this->hydrateDocumentUrls($document), $documents)